src/Handler/ExecuteQuery/ExecuteQueryHandler.php

```diff
@@ -0,0 +1,138 @@
+            throw new InvalidArgumentException('Restriction is required for query execution.');
+        }
+
+        if ($datasetName === null) {
+            throw new InvalidArgumentException('Dataset name is required for query execution.');
+        }
+        // resolve query timeout
+        $queryTimeout = self::DEFAULT_QUERY_TIMEOUT_SECONDS;
+        if (is_int($command->getTimeout()) && $command->getTimeout() > 0) {
+            $queryTimeout = $command->getTimeout();
+        }
+
+        // crate service account key
+        $iamService = $this->clientManager->getIamClient($credentials);
+        $serviceAccKeysService = $iamService->projects_serviceAccounts_keys;
+
+        $serviceAccResourceName = sprintf(
+            'projects/%s/serviceAccounts/%s',
+            $restriction->getProjectId(),
+            $restriction->getServiceAccountEmail(),
+        );
+
+        // create new service account key
+        $createServiceAccountKeyRequest = new CreateServiceAccountKeyRequest();
+        $createServiceAccountKeyRequest->setPrivateKeyType(CreateWorkspaceHandler::PRIVATE_KEY_TYPE);
+        $serviceAccount = $iamService->projects_serviceAccounts->get($serviceAccResourceName);
+        [$privateKey, $publicPart, $keyName] = $iamService->createKeyFileCredentials($serviceAccount);
+
+        try {
+            // initialize BigQuery client with the service account key
+            $bqClient = $this->clientManager->getBigQueryClient(
+                $runtimeOptions->getRunId(),
+                new GenericBackendCredentials([
+                    'host' => $credentials->getHost(),
+                    'principal' => $publicPart,
+                    'secret' => $privateKey,
+                    'port' => $credentials->getPort(),
+                    'meta' => $credentials->getMeta(),
+                ]),
+            );
+            // prepare query job configuration
+            $dataset = $bqClient->dataset($datasetName);
+            $queryJobConfiguration = $bqClient->query(
+                $command->getQuery(),
+                [
+                    'configuration' => [
+                        'jobTimeoutMs' => $queryTimeout * 1000,
+                    ],
+                ]
+            )->defaultDataset($dataset);
+
+            // execute the query
+            $result = $bqClient->runQuery($queryJobConfiguration);
+        } catch (Throwable $e) {
+            $this->internalLogger->error($e->getMessage());
+            return new ExecuteQueryResponse([
+                'status' => ExecuteQueryResponse\Status::Error,
+                'message' => DecodeErrorMessage::getErrorMessage($e),
+            ]);
+        } finally {
+            // delete the service account key
+            $serviceAccKeysService->delete($keyName);
+        }
+
+        $rows = array_map(fn($r) => new ExecuteQueryResponse\Data\Row([
+            'fields' => $r,
+        ]), iterator_to_array($result));
+
+        return new ExecuteQueryResponse([
```

Comment:
vime co to vrati pro treba INSERT query? jestli nejsme schopni v nejake message najit treba pocet ovlivnenych radku. A to bychom pak mohli vratit

Reply:
Jo to sem zkůšal a nic to nevrací.
Ide to lehko otestovat
stačí nad ten return přidat 
```
        $this->internalLogger->info(
            json_encode($result->info()),
        );
        $this->internalLogger->info(
            json_encode($result->job()),
        );
        $this->internalLogger->info(
            json_encode($result->identity()),
        );
```
pak sa to objeví v ./logs složce

Ale přidám tam message s id té query.